conan/tools/cmake/cmakedeps/templates/config.py

```diff
@@ -28,7 +28,7 @@ def context(self):
         targets_include += "{}Targets.cmake".format(self.file_name)
         return {"is_module": self.generating_module,
                 "version": self.conanfile.ref.version,
-                "file_name": self.file_name,
+                "file_name": self.legacy_variable_prefix or self.file_name,
```

Comment:
This is going to have unwanted changes in the files generated by CMakeDeps when someone declares this in a recipe it will affect more things than the ones this PR was supposed to change.